app/models/concerns/course_component_query_concern.rb

```diff
@@ -31,10 +31,48 @@ def user_enabled_components
     end
   end
 
+  # Set component's `enabled` key to be either true or false.
+  #
+  # @param [Symbol|String] key Component key
+  # @param [Boolean] value true if component is to be enabled, false otherwise.
+  def unsafe_set_component_enabled(key, value)
+    component_setting(key).enabled = value
+  end
+
+  # Set component's `enabled` key only if it is disableable
+  def set_component_enabled(key, value)
```

Comment:
Would something like `course.enable_component` and `course.disable_component` be more naturally readable?

Reply:
I decided against it because the cost of the readability at the call sites is a proliferation of methods in the concern:

```ruby
course.enable_component(key)
course.enable_component!(key)
course.enable_component(key, force: true) # unsafe
course.disable_component(key)
course.disable_component!(key)
course.disable_component(key, force: true) # unsafe
```

We could stick to `unsafe_set_component_enabled` for the unsafe versions I guess, but we would not have that pristine neatness anymore, which to me negates a bit of the benefit of having those methods, since now we have to juggle the two groups of APIs.

Do you think it is worth the change?

Reply:
How about naming it `set_component_enabled_key` then?

It's kind of ok the way it is, but seeing `course.set_component_enabled(disabled_key, false)`, with `enabled` and `disabled_key` in the same line was a little incongruent.

Reply:
How about if I changed it to 
```ruby
course.set_component_enabled(disabled_component.key, false)
```
Would that be clearer?

If we rename the method, would `set_component_enabled_boolean` be better?

Reply:
Let's do both?